mpqp/core/circuit.py

```diff
@@ -165,7 +164,17 @@ def __init__(
             self.add(deepcopy(data))
 
     def __eq__(self, value: object) -> bool:
-        return dumps(self) == dumps(value)
```

Comment:
can you provide me with an example where this didn't work (for my culture)

Reply:
The problem came from somewhere else but I had kept the change. I discard the change and all the tests passed.